### DISCUSSION TITLE
How to create vertices and attach to them when link-to-link connection?

### DISCUSSION THREAD

**Discussion by 0niel** [clientIO/joint#2187] [Q&A] [answered]
### Introduction

Hello! I am creating a cool open source solution for educational institutions to create a svg indoor maps using graph and **jointjs**!

![image](https://github.com/clientIO/joint/assets/51058739/570da968-8890-4e7a-9fc4-fa1f33c54024)

I have a problem when I attach a link to another link. When joining, I get an anchor that has this structure:
```
"id": "0269e748-5ddd-4060-9bab-80aae78c0024",
"anchor": {
    "name": "connectionRatio",
    "args": {
      "ratio": 0.26327396363743777
    }
  }
```

I would like that when joining, a new **vertex** is created and the link is attached to it. This is important when using my algorithm. I also use an undirected graph.

Now I'm working on strange solutions, for example, I can use `path.divideAt` to divide the link path and create a vertex there somehow. 
```ts
paper.on("link:connect", function(linkView, evt, elementViewConnected, magnet, arrowhead) {
      if (elementViewConnected.model.isLink()) {

        const path = elementViewConnected.path;
        elementViewConnected.path = path.divideAt(ratioHere);
        
        ...
      }
});
```

Could you help me solve the problem?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
A custom link anchor would be required. Please see this [JSFiddle](https://jsfiddle.net/kumilingus/3p6aok1h/).
